pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -262,6 +262,46 @@ func (a *ApplicationBackupController) handle(ctx context.Context, backup *stork_
 			if !canDelete {
 				return nil
 			}
+			// run the post exec rules if the backup is in volume stage and execRulesCompleted is not completed
+			if _, ok := a.execRulesCompleted[string(backup.UID)]; !ok && backup.Status.Stage == stork_api.ApplicationBackupStageVolumes {
+				if terminationChannels, ok := a.terminationChannels[string(backup.UID)]; ok {
+					for _, channel := range terminationChannels {
+						logrus.Infof("Sending termination commands to kill pre-exec pod in non-kdmp driver path")
+						channel <- true
+					}
+				}
+				if backup.Spec.PostExecRule != "" {
+					log.ApplicationBackupLog(backup).Infof("Starting post-exec rule in delete path")
+					err = a.runPostExecRule(backup)
+					if err != nil {
+						message := fmt.Sprintf("Error running PostExecRule: %v", err)
+						log.ApplicationBackupLog(backup).Errorf(message)
+						a.recorder.Event(backup,
+							v1.EventTypeWarning,
+							string(stork_api.ApplicationBackupStatusFailed),
+							message)
+						return fmt.Errorf("%v", message)
+					}
+					a.execRulesCompleted[string(backup.UID)] = true
+				}
+			}
+
+			// Delete post-exec rule CR if any
+			if len(backup.Spec.PostExecRule) != 0 && backup.Annotations[utils.PxbackupAnnotationCreateByKey] == utils.PxbackupAnnotationCreateByValue && backup.Annotations["portworx.io/backup-by"] != "backup-schedule" && backup.Annotations["portworx.io/backup-by"] != "restore" {
+				log.ApplicationBackupLog(backup).Info("Delete post-exec rule CR")
+				//uid := backup.GetAnnotations()["portworx.io/backup-uid"]
+				//ruleCrName := getCrNameWithUID(backup.Spec.PostExecRule, uid)
+
+				err := storkops.Instance().DeleteRule(backup.Spec.PostExecRule, backup.Namespace)
+				log.ApplicationBackupLog(backup).Infof("Error value: %v", err)
+
+				if err != nil && !k8s_errors.IsNotFound(err) {
+					log.ApplicationBackupLog(backup).Infof("Error while deleting post exec rule: %v", err)
```

Comment:
Let us have all the error msg and debug in the context to refer to "during cancel backup" or some other statement.